esx_service/utils/vmdk_utils.py

```diff
@@ -44,18 +46,23 @@ def get_datastores():
     #  We are connected to ESX so childEntity[0] is current DC/Host
     ds_objects = \
       si.content.rootFolder.childEntity[0].datastoreFolder.childEntity
-    datastores = [(d.info.name,
+    datastores = [[d.info.name,
                    os.path.split(d.info.url)[1],
-                   os.path.join(d.info.url, 'dockvols'))
+                   os.path.join(d.info.url, 'dockvols'), True]
                   for d in ds_objects]
+    for d in ds_objects:
+       ds = [x for x in datastores if x[0] == d.info.name]
+       if d.summary.type == 'VMFS':
+          ds[0][3] = d.info.vmfs.local
```

Comment:
The VMODL has a boolean for VMFS datastores called "local" which is
true/false depending on whether the device is local or not.


On Tue, Jul 12, 2016 at 9:02 PM, Mark Sterin <notifications@github.com>
wrote:

> In esx_service/utils/vmdk_utils.py
> <https://github.com/vmware/docker-volume-vsphere/pull/530#discussion_r70463188>
> :
>
> >                    for d in ds_objects]
> > +    for d in ds_objects:
> > +       ds = [x for x in datastores if x[0] == d.info.name]
> > +       if d.summary.type == 'VMFS':
> > +          ds[0][3] = d.info.vmfs.local
>
> I am not sure why "VMFS "indicates "local". What if it's just an array LUN
> exposed via iSCSI and formatted as VMFS (?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/vmware/docker-volume-vsphere/pull/530/files/ee6e8a6a3c7329bc516f2232ccddab472f8e7fb8#r70463188>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/APHseAm9T0GwTCBrJi3MMIJR-LsbvanOks5qU7OigaJpZM4JKRmT>
> .
>